tests/integration/auth.spec.coffee

```diff
@@ -0,0 +1,144 @@
+m = require('mochainon')
+
+{ resin, sdkOpts, credentials, givenLoggedInUser } = require('./setup')
+
+describe 'SDK authentication', ->
+
+	describe 'when not logged in', ->
+
+		beforeEach ->
+			resin.auth.logout()
+
+		describe 'resin.auth.isLogged()', ->
+
+			it 'should eventually be false', ->
+				promise = resin.auth.isLoggedIn()
+				m.chai.expect(promise).to.eventually.be.false
+
+		describe 'resin.auth.whoami()', ->
+
+			it 'should eventually be undefined', ->
+				promise = resin.auth.whoami()
+				m.chai.expect(promise).to.eventually.be.undefined
+
+		describe 'resin.auth.logout()', ->
+
+			it 'should not be rejected', ->
+				promise = resin.auth.logout()
+				m.chai.expect(promise).to.not.be.rejected
+
+		describe 'resin.auth.authenticate()', ->
+
+			it 'should eventually be a valid token given valid credentials', ->
+				resin.auth.authenticate(credentials).then(resin.token.isValid).then (isValid) ->
+					m.chai.expect(isValid).to.be.true
+
+			it 'should not save the token given valid credentials', ->
+				resin.auth.authenticate(credentials).then ->
+					promise = resin.auth.isLoggedIn()
+					m.chai.expect(promise).to.eventually.be.false
+
+			it 'should be rejected given invalid credentials', ->
+				promise = resin.auth.authenticate
+					email: 'helloworld@resin.io',
+					password: 'asdfghjkl'
+
+				m.chai.expect(promise).to.be.rejectedWith('Unauthorized')
+
+		describe 'resin.auth.getEmail()', ->
+
+			it 'should be rejected with an error', ->
+				promise = resin.auth.getEmail()
+				m.chai.expect(promise).to.be.rejected
+					.and.eventually.have.property('code', 'ResinNotLoggedIn')
+
+		describe 'resin.auth.getUserId()', ->
+
+			it 'should be rejected with an error', ->
+				promise = resin.auth.getUserId()
+				m.chai.expect(promise).to.be.rejected
+					.and.eventually.have.property('code', 'ResinNotLoggedIn')
+
+		describe 'resin.auth.register()', ->
+
+			beforeEach ->
+				resin.auth.login
+					email: credentials.register.email
+					password: credentials.register.password
+				.then(resin.auth.getUserId)
+				.then (userId) ->
+					return resin.request.send
+						method: 'DELETE'
+						url: "/ewa/user(#{userId})"
+						baseUrl: sdkOpts.apiUrl
+					.then(resin.auth.logout)
+				.catch (error) ->
```

Comment:
This can be
```coffee
.catchReturn(message: 'Request error: Unauthorized', undefined)
```
or
```coffee
.catch(message: 'Request error: Unauthorized', ->)